### PR TITLE
feat: rtl tests refactoring

### PR DESCRIPTION
**Pull request by PavelKirvelTextkernel**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

# Checklist
- [ ] The implementation has been manually tested and complies with Textkernel [browser support guidelines](https://textkernel.com/browser-support/)
- [ ] The implementation complies with [accessibility](CONTIRBUTING.md#accessibility) standards.
- [ ] The component has a [displayName](CONTRIBUTING.md#display-names) defined.
- [ ] The component comes with a detailed [PropTypes](CONTRIBUTING.md#component-props) (and defaultProps) definition.
- [ ] Component PropTypes are sufficiently described / documented.
- [ ] There is [a story](CONTRIBUTING.md#component-showcases) in Storybook.

Task link: https://textkernel.atlassian.net/browse/ONEUI-375

### CODE REVIEW

**Review by eszthoff — CHANGES_REQUESTED** [review]

**Comment by PavelKirvelTextkernel** [comment]
https://textkernel.atlassian.net/browse/ONEUI-375

File: TESTING_LIBRARY_MIGRATION.md [hunk 1]
```diff
@@ -114,11 +116,12 @@ expect(component).toMatchSnapshot();
...
 ```ts
-import { fireEvent, render, screen } from 'react-test-renderer';
+import { render, screen } from 'react-test-renderer';
+import userEvent from '@testing-library/user-event';
 
-const component = render(<MyComponent />);
-fireEvent.change(screen.getByDisplayValue(''), { target: { value: 'Utrecht' } });
-expect(component.asFragment()).toMatchSnapshot();
+const { asFragment } = render(<MyComponent />);
+await userEvent.type(screen.getByDisplayValue(''), 'Utrecht');
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply TESTING_LIBRARY_MIGRATION.md:123]: Also, I thought one should call `setup` for the user first. So:
`const user = userEvent.setup()`
see: https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: TESTING_LIBRARY_MIGRATION.md [hunk 2]
```diff
@@ -96,12 +98,12 @@ expect(component).toMatchSnapshot();
 ```ts
 import { render, screen } from 'react-test-renderer';
 
-const component = render(<MyComponent />);
+const { asFragment } = render(<MyComponent />);
 expect(screen.getByText('value')).toBeDefined();
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply TESTING_LIBRARY_MIGRATION.md:102]: The idea was to provide different methods for this doc

[the remark this answers is not in the document]
Comment (eszthoff) [reply TESTING_LIBRARY_MIGRATION.md:102]: Sorry, I don't get it. The before and the after seems to be unrelated. Maybe an equivalent test should be added to the `Before` section, no?

[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply TESTING_LIBRARY_MIGRATION.md]: :+1:

File: TESTING_LIBRARY_MIGRATION.md [hunk 3]
```diff
@@ -89,19 +91,20 @@ expect(button).toBeInTheDocument();
 import { shallow } from 'enzyme';
 
 const component = shallow(<MyComponent />);
+expect(wrapper.find('value')).toHaveLength(1);
```
Comment (eszthoff) [inline TESTING_LIBRARY_MIGRATION.md:94]:
```suggestion
expect(wrapper.find('Button')).toHaveLength(1);
```

File: TESTING_LIBRARY_MIGRATION.md [hunk 4]
```diff
@@ -89,19 +91,20 @@ expect(button).toBeInTheDocument();
...
 
 #### After ✅:
 ```ts
 import { render, screen } from 'react-test-renderer';
 
-const component = render(<MyComponent />);
-expect(screen.getByText('value')).toBeDefined();
-expect(component.asFragment()).toMatchSnapshot();
+const { asFragment } = render(<MyComponent />);
+expect(screen.getByRole('value')).toBeInTheDocument();
```
Comment (eszthoff) [inline TESTING_LIBRARY_MIGRATION.md:103]:
```suggestion
expect(screen.getByRole('button')).toBeInTheDocument();
```

File: TESTING_LIBRARY_MIGRATION.md [hunk 5]
```diff
@@ -89,19 +91,20 @@ expect(button).toBeInTheDocument();
...
+expect(asFragment()).toMatchSnapshot();
```
Comment (eszthoff) [inline TESTING_LIBRARY_MIGRATION.md:104]:
```suggestion
```

File: TESTING_LIBRARY_MIGRATION.md [hunk 6]
```diff
@@ -89,19 +91,20 @@ expect(button).toBeInTheDocument();
 import { shallow } from 'enzyme';
 
 const component = shallow(<MyComponent />);
+expect(wrapper.find('value')).toHaveLength(1);
 expect(component).toMatchSnapshot();
```
Comment (eszthoff) [inline TESTING_LIBRARY_MIGRATION.md:95]:
```suggestion
```

File: TESTING_LIBRARY_MIGRATION.md (hunk 3 above)
Comment (PavelKirvelTextkernel) [reply TESTING_LIBRARY_MIGRATION.md:94]: :+1:

File: TESTING_LIBRARY_MIGRATION.md (hunk 4 above)
Comment (PavelKirvelTextkernel) [reply TESTING_LIBRARY_MIGRATION.md:103]: :+1:

File: TESTING_LIBRARY_MIGRATION.md (hunk 5 above)
Comment (PavelKirvelTextkernel) [reply TESTING_LIBRARY_MIGRATION.md:104]: :+1:

File: TESTING_LIBRARY_MIGRATION.md (hunk 6 above)
Comment (PavelKirvelTextkernel) [reply TESTING_LIBRARY_MIGRATION.md:95]: :+1:

**Pull request merged**